src/common/transport/transport-client.js

```diff
@@ -0,0 +1,80 @@
+import { TransportClientError } from '../errors';
+import rp from 'request-promise';
```

Comment:
Do you prefer to have a separate transporter for node and the browser, or just stick with one that works for both? Also, if this is going to work on a browser shouldn't we stick with babel to compile down to ES5? ES6 support across browsers is still behind what node supports.